vmas/scenarios/navigation_obstacles.py

```diff
@@ -0,0 +1,412 @@
+            )
+            agent.pos_rew = torch.zeros(batch_dim, device=device)
+            agent.agent_collision_rew = agent.pos_rew.clone()
+            world.add_agent(agent)
+
+            # Add goals
+            goal = Landmark(
+                name=f"goal {i}",
+                collide=False,
+                color=color,
+            )
+            world.add_landmark(goal)
+            agent.goal = goal
+
+        # Add in the map obstacles
+        indices = torch.nonzero(self.map == 1)
+        for coord in indices:
+            obstacle = Landmark(
+                name = f"Obstacle at {coord}",
+                collide = True,
+                shape = Box(self.obstacle_dim, self.obstacle_dim),
+                collision_filter = lambda e: isinstance(e, Agent)
+            )
+            world.add_landmark(obstacle)   
+
+        self.pos_rew = torch.zeros(batch_dim, device=device)
+        self.final_rew = self.pos_rew.clone()
+
+        return world 
+    
+    def parse_map(self, map_name: str):
+        # Load in map xml
+        f = os.path.dirname(__file__)
+        map = ET.parse(f'{f}/../../maps/{map_name}.xml')
+        map_root = map.getroot()
+
+        # Extract grid dimensions
+        width = int(map_root.find('.//width').text)
+        height = int(map_root.find('.//height').text)
+
+        # Initialize a numpy array to store the grid
+        grid = torch.zeros((height, width), dtype=torch.int, device=self.device)
+
+        # Find all row elements and populate the numpy array
+        row_elements = map_root.findall('.//row')
+        for i, row_element in enumerate(row_elements):
+            row_data = row_element.text.split()
+            for j, value in enumerate(row_data):
+                grid[height - 1 - i, j] = int(value)
+        return grid, (0, width), (0, height)
+    
+    def reset_world_at(self, env_index: int = None):
+        # TODO [Shalin]: env_index is randomly not None during training, which breaks resetting
+        env_index = None
+
+        # store positions occupied by obstacles
+        occupied_obstacles = torch.zeros((self.world.batch_dim, len(self.world.landmarks[self.n_agents:]), 2), 
+                                         dtype=torch.float32,
+                                         device=self.world.device)
+
+        # NOTE: this needs to happen every reset to be compatible with BaseScenario reset
+        for i, pos in enumerate(torch.nonzero(self.map == 1)):
+            self.world.landmarks[i+self.n_agents].set_pos(
+                torch.tensor(
+                    [pos[1], pos[0]],
+                    dtype=torch.float32,
+                    device=self.device
+                ),
+                batch_index = env_index
+            )
+            occupied_obstacles[:, i] = self.world.landmarks[i+self.n_agents].state.pos
+
+        if self.map_name == "alcove":
+            # spawn first agent position in left side of map (with noise)
```

Comment:
Rather than hardcoding spawn/goal positions for alcove, would it be possible to encode this in the map .xmls? Something like 
```
X = obstacle
. = free space
<int>s = agent <int> spawn
<int>g = agent <int> goal
<int1>s<int2>g = agent <int1> spawn, agent <int2> goal
```

So for instance alcove would look like:
![Screenshot 2024-01-16 at 1 55 11 PM](https://github.com/GT-STAR-Lab/VectorizedMultiAgentSimulator/assets/30275369/c6858bb8-796f-4132-86a5-939368b9b28b)

Let me know if this is impossible for some reason. I think being able to specify it from the files will make it easier to run experiments / set up curriculum learning. (Also, feel free to use a different encoding scheme if you think of something more compact.)